now/executor/indexer/elastic/elastic_indexer.py

```
@@ -88,38 +85,42 @@ def __init__(
         }
         self.es_config = es_config or {'verify_certs': False}
         self.es_mapping = es_mapping or self.generate_es_mapping()
-        self.setup_elastic_server()
-        self.es = Elasticsearch(hosts=self.hosts, **self.es_config, ssl_show_warn=False)
-        wait_until_cluster_is_up(self.es, self.hosts)
-
+        self.es = Elasticsearch(
+            hosts=self.hosts,
+            api_key=self.api_key,
+            **self.es_config,
+            ssl_show_warn=False,
+        )
+        self._do_health_check()
         if not self.es.indices.exists(index=self.index_name):
             self.es.indices.create(index=self.index_name, mappings=self.es_mapping)
+        else:
+            self.es.indices.put_mapping(index=self.index_name, body=self.es_mapping)
 
-    def setup_elastic_server(self):
-        try:
-            if "K8S_NAMESPACE_NAME" in os.environ:
-                data_path = f'/data/{os.environ["K8S_NAMESPACE_NAME"]}'
-                subprocess.run(["chmod", "-R", "0777", data_path])
-                self.configure_elastic(
-                    data_path,
-                    '/usr/share/elasticsearch/config/elasticsearch.yml',
+    def _check_env_vars(self):
+        while not all(
+            var in os.environ for var in ['ES_HOSTS', 'ES_INDEX_NAME', 'ES_API_KEY']
+        ):
+            timeout_counter = 0
+            if timeout_counter < TIMEOUT:
+                timeout_counter += 5
+                self.logger.info('Environment variables not set yet. Waiting...')
+                sleep(5)
```

why do we need to wait for this? in other words, what could change while waiting? maybe we should throw an error immediately and tell the user (us) to set envs

---

The operator needs to provision the resource first, and then set these variables in our container. This may take a little while, so we wait :)